src/main/scala/mesosphere/marathon/core/election/impl/CuratorElectionService.scala

```diff
@@ -125,6 +126,25 @@ class CuratorElectionService(
           CurrentRuntime.asyncExit()(scala.concurrent.ExecutionContext.global)
           false
         }
+      }).zookeeperFactory(new ZookeeperFactory {
+        val delegate = new DefaultZookeeperFactory
+        override def newZooKeeper(
+          connectString: String,
+          sessionTimeout: Int,
+          watcher: Watcher,
+          canBeReadOnly: Boolean): ZooKeeper =
+
+          try {
+            delegate.newZooKeeper(connectString, sessionTimeout, watcher, canBeReadOnly)
+          } catch {
+            // We may see exceptions here, for example, when none of the ZK hosts are resolvable at Marathon startup;
+            // in this case Curator may fail with some background exception and Marathon never offers leadership.
+            case NonFatal(e) =>
+              // TODO(jdef) should we really be committing suicide here? what about retries instead?
+              log.error("ZooKeeper initialization failed - Committing suicide")
+              CurrentRuntime.asyncExit()(scala.concurrent.ExecutionContext.global)
+              null
```

Comment:
I'd like to churn on this a bit more. My inclination is to do the resolution ourselves in a Retry.blocking call, but this gets weird because of how java caches DNS. It seems wrong to me to change their factory to get around the bug instead of giving it a list of IPs